umm-spec-lib/resources/json-schemas/subscription/umm/v1.0/umm-sub-json-schema.json

```diff
@@ -0,0 +1,45 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "UMM-Sub",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "Name": {
+      "description": "The name of a subscription.",
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 80
+    },
+    "SubscriberId": {
+      "description": "The userid of the subscriber.",
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 30
+    },
+    "EmailAddress": {
+      "description": "The email address of the subscriber.",
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 255
+    },
+    "CollectionConceptId": {
+      "description": "The collection concept id of the granules subscribed.",
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 255 
+    },
+    "Query": {
+      "description": "The search query for the granules that matches the subscription.",
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 40000 
```

Comment:
Not sure how big this maxLength could be.

Reply:
https://stackoverflow.com/questions/417142/what-is-the-maximum-length-of-a-url-in-different-browsers

This is in search-app/docs/api.md:
The Maximum URL Length supported by CMR is indirectly controlled by the Request Header Size setting in Jetty which is set to 1MB. This translates to roughly 500k characters, however it is recommended that any GET request be limited to 6,000 characters, and in a web browser the recommended length is no longer than 2000 characters. Clients using the Search API with query parameters should be careful not to exceed this limit or they will get an HTTP response of 413 FULL HEAD. If a client expects that the query url could be extra long so that it exceeds 6k characters, they should use the POST API for searching.

for now 40,000 is more than sufficient.

Reply:
@eereiter  Thank you so much for the info!